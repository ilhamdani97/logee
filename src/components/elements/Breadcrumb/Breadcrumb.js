import React from 'react';
import styles from './styles.scoped.css';
import PropTypes from 'prop-types';
import compareProps from '../../../utils/compareProps';
import { Link } from 'react-router-dom';

function Breadcrumb({ paths }) {
  return (
    <section className={styles.root}>
      <ul>
        {paths && paths.map((path, idx) => {
          if(idx === 0){
            return (
              <li key={idx}>
                <Link to={path.location}>
                  {path.name}
                </Link>
              </li>
            );
          } else {
            return (
              <React.Fragment key={idx}>
                <span>/</span>
                <li>
                  <Link to={path.location}>
                    {path.name}
                  </Link>
                </li>
              </React.Fragment>
            );
          }
        })}

      </ul>
    </section>
  );
}

export default React.memo(Breadcrumb, compareProps);

Breadcrumb.defaultProps = {
  paths: []
};

Breadcrumb.propTypes = {
  paths: PropTypes.array
};
