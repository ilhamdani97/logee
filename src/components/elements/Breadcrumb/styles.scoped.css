.root {
  margin-bottom: 0.1875rem;
}

.root > ul {
  display: flex;
  color: #a0a4a8;
}

.root > ul > span {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.root > ul > li,
.root > ul > li > a {
  font-size: 0.75rem;
  line-height: 1rem;
  text-transform: capitalize;
}

.root > ul > li > a:hover {
  text-decoration: underline;
}

.root > ul > li:last-child > a {
  text-decoration: none;
  pointer-events: none;
  cursor: not-allowed;
}
