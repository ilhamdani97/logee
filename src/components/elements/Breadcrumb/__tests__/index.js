import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Breadcrumb from '../Breadcrumb';

const breadCrumbProps = [
  {
    name: `impor`,
    location: `/${'jict'}/listingordertruck?${'type=EXPORT'}`,
  },
  {
    name: 'List Pesanan Truk',
    location: `/${'jict'}/listingordertruck?${'type=EXPORT'}`,
  },
];

describe('src/components/elements/Breadcrumb', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Breadcrumb paths={breadCrumbProps} />);
    expect(tree).toMatchSnapshot();
  });
});
