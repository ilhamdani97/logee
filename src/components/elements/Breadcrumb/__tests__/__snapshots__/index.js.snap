// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Breadcrumb render 1`] = `
<section>
  <ul>
    <li>
      <Link
        to="/jict/listingordertruck?type=EXPORT"
      >
        impor
      </Link>
    </li>
    <React.Fragment>
      <span>
        /
      </span>
      <li>
        <Link
          to="/jict/listingordertruck?type=EXPORT"
        >
          List Pesanan Truk
        </Link>
      </li>
    </React.Fragment>
  </ul>
</section>
`;
